test/test-avx512-util.cpp

```diff
@@ -8,27 +8,37 @@
 #include <vector>
 
 #include "gtest/gtest.h"
-#include "test-util.hpp"
+#include "test-util-avx512.hpp"
 #include "util/avx512-util.hpp"
+#include "util/cpu-features.hpp"
 
 namespace intel {
 namespace hexl {
 
 #ifdef HEXL_HAS_AVX512DQ
 
 TEST(AVX512, ExtractValues) {
+  if (!has_avx512dq) {
+    return;
+  }
```

Comment:
Is this check really necessary? The `ifdef` on line 18 means these tests shouldn't run if not on an AVX512 machine.
Also should the desired behaviour be AVX512 specific tests pass on non-AVX512 machines?

Reply:
Anything AVX less than AVX512DQ will not give any performance gains ... and in some cases potentially degrade it.

Reply:
Today, `#ifdef HEXL_HAS_AVX512DQ` is defined iff the processor and compiler support the AVX512DQ instruction set. In the future, it will be defined iff the compiler supports the AVX512DQ instruction set. So, today this change doesn't make a difference, it's just a preparation for a future PR in which `HEXL_HAS_AVX512DQ` is defined on processors without AVX512DQ support.